go/arrow/array/dictionary.go

```diff
@@ -1004,6 +1004,10 @@ func (b *dictionaryBuilder) AppendIndices(indices []int, valid []bool) {
 	}
 }
 
+func (b *dictionaryBuilder) DictionarySize() int {
+	return b.memoTable.Size()
+}
+
```

Comment:
We should probably add this to the `DictionaryBuilder` interface, yea?

Reply:
Yes, done in https://github.com/apache/arrow/pull/39521/commits/98b970fb09e6ccf90f174f8624049bbd11d59ac8